src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/profile/Profile.java

```diff
@@ -50,39 +43,34 @@ private Profile() {
       blavf.setWeight(c, 0d);
     }
     this.linearAvf = blavf.build();
-    this.floorAreaWeightRange = null;
-    this.nbBedroomsWeightRange = null;
-    this.nbSleepingWeightRange = null;
-    this.nbBathroomsWeightRange = null;
-    this.terraceWeightRange = null;
-    this.floorAreaTerraceWeightRange = null;
-    this.wifiSubjectiveValueWeightRange = null;
-    this.pricePerNightSubjectiveValueWeightRange = null;
-    this.nbMinNightSubjectiveValueWeightRange = null;
-    this.teleSubjectiveValueWeightRange = null;
+
+    this.rangeMap = new EnumMap<>(Criterion.class);
+    for (Criterion c : Criterion.values()) {
+      this.rangeMap.put(c, null);
+    }
   }
 
   /**
-   * This function allows the user to clone an object Profile
+   * This function allows the user to clone an object Profile This function allows you to indicate
+   * which value should be cloned. Thus, each parameter not present will be cloned from the current
+   * Profile To modify the profile during the clone, just give it the desired parameters
    *
-   * @return an object Profile
+   * @param rangeMap corresponds to the range associated with the <code>Profile</code>. In the case
+   *     where an <code>EnumMap</code> is present, it does not have to contain all the necessary
+   *     keys
+   * @param lavf corresponds to the LinearAVF associated with the Profile. In case it is not
+   *     present, we clone the LinearAVF from current <code>Profile</code>
+   * @return Profile modified (if Optional are present)
    */
-  private Profile cloneProfile() {
-
+  private Profile clone(
+      Optional<EnumMap<Criterion, Range<Double>>> rangeMap, Optional<LinearAVF> lavf) {
```

Comment:
Attention, il y a un warning ici. Il faut renommer rangeMap pour le supprimer.

Reply:
FIx in 8fb556a04f0d9d45e0ec39ce9706e7cb06409eb1